### DISCUSSION TITLE
gatsby-plugin-mdx: failing to use remark-mdx-frontmatter as remark plugin

### DISCUSSION THREAD

**Discussion by openscript** [gatsbyjs/gatsby#36104] [Help] [answered]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

After configuring [`remark-mdx-frontmatter`](https://github.com/remcohaszing/remark-mdx-frontmatter) the `gatsby-plugin-mdx` doesn't work anymore and subsequent queries depending on mdx failing:

```
ERROR #85923  GRAPHQL

There was an error in your GraphQL query:

Cannot query field "mdx" on type "Query".
```

https://github.com/openscript/gatsby-mdx-remark-frontmatter-plugin/blob/c0ecdbd99bfe5ecd555a05c3e76cd0298dae993d/gatsby-config.ts#L57

### Reproduction Link

https://github.com/openscript/gatsby-mdx-remark-frontmatter-plugin

### Steps to Reproduce

1. Start the reproduction project with `yarn start`

### Expected Result

The plugin `remark-mdx-frontmatter` is loaded and executed by mdx.

### Actual Result

`gatsby-plugin-mdx` doesn't work anymore.

### Environment

```shell
System:
    OS: Linux 5.18 Debian GNU/Linux 11 (bullseye) 11 (bullseye)
    CPU: (8) x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 5.1.4 - /bin/bash
  Binaries:
    Node: 18.0.0 - /tmp/yarn--1657483014690-0.4392544167926853/node
    Yarn: 1.22.18 - /tmp/yarn--1657483014690-0.4392544167926853/yarn
    npm: 8.6.0 - /usr/local/bin/npm
  Languages:
    Python: 3.9.2 - /usr/bin/python
  npmPackages:
    gatsby: 4.18.0 => 4.18.0
    gatsby-plugin-catch-links: 4.18.0 => 4.18.0
    gatsby-plugin-emotion: 7.18.0 => 7.18.0
    gatsby-plugin-i18n-l10n: 4.18.5 => 4.18.5
    gatsby-plugin-image: 2.18.0 => 2.18.0
    gatsby-plugin-manifest: 4.18.0 => 4.18.0
    gatsby-plugin-mdx: 3.18.0 => 3.18.0
    gatsby-plugin-offline: 5.18.0 => 5.18.0
    gatsby-plugin-react-helmet: 5.18.0 => 5.18.0
    gatsby-plugin-sharp: 4.18.0 => 4.18.0
    gatsby-remark-autolink-headers: 5.18.0 => 5.18.0
    gatsby-remark-copy-linked-files: 5.18.0 => 5.18.0
    gatsby-remark-images: 6.18.0 => 6.18.0
    gatsby-source-filesystem: 4.18.0 => 4.18.0
    gatsby-transformer-sharp: 4.18.0 => 4.18.0
```

**Comment by LekoArts** [comment, marked as answer]
The package README (https://github.com/remcohaszing/remark-mdx-frontmatter) states:

> This package depends on the AST output by [remark-frontmatter](https://github.com/remarkjs/remark-frontmatter)

Since we don't use this package, it can't work. There should also no need to use this package, why did you want to use it?

**Reply by openscript** [reply]
Thank you for your answer. I wished to reuse frontmatter inside the markdown like:

```
---
title: This is the title
---

# {title}
```

Can this be achieved in another way than add the title in the template component by querying the frontmatter title?

I understand that it's difficult to make it work with `remark-mdx-frontmatter` due the AST manipulation of `remark-frontmatter`.

**Reply by LekoArts** [reply]
You can use `{props.pageContext.frontmatter.title}`, see https://www.gatsbyjs.com/docs/mdx/writing-pages/#using-frontmatter-in-mdx

**Reply by openscript** [reply]
Ah, cool! Sorry I didn't see this before. :)
